File: accommodation_service/src/main/java/huy/project/accommodation_service/core/port/IImagePort.java
package huy.project.accommodation_service.core.port;

import huy.project.accommodation_service.core.domain.entity.ImageEntity;

import java.util.List;

public interface IImagePort {
    List<ImageEntity> saveAll(List<ImageEntity> images);
    List<ImageEntity> getImagesByEntityIdAndType(Long entityId, String entityType);
    void deleteImagesByEntityIdAndType(Long entityId, String entityType);
}

File: location_service/LocationService.Core/Port/IImagePort.cs
using LocationService.Core.Domain.Entity;

namespace LocationService.Core.Port
{
    public interface IImagePort
    {
        Task<List<ImageEntity>> CreateImagesAsync(List<ImageEntity> images);
        Task<List<ImageEntity>> GetImagesByEntityIdAndEntityType(long entityId, string entityType);
    }
}